tensorflow_addons/seq2seq/sampler.py

```diff
@@ -29,25 +29,34 @@
 class Sampler(metaclass=abc.ABCMeta):
     """Interface for implementing sampling in seq2seq decoders.
 
-    Sampler instances are used by `BasicDecoder`. The normal usage of a sampler
-    is like below:
-
-    ```python
-    sampler = Sampler(init_args)
-    (initial_finished, initial_inputs) = sampler.initialize(input_tensors)
-    cell_input = initial_inputs
-    cell_state = cell.get_initial_state(...)
-    for time_step in tf.range(max_output_length):
-        cell_output, cell_state = cell(cell_input, cell_state)
-        sample_ids = sampler.sample(time_step, cell_output, cell_state)
-        (finished, cell_input, cell_state) = sampler.next_inputs(
-            time_step, cell_output, cell_state, sample_ids)
-        if tf.reduce_all(finished):
-            break
-    ```
-
-    Note that the input_tensors should not be fed to the Sampler as __init__()
-    parameters. Instead, they should be fed by decoders via initialize().
+    Sampler classes implement the logic of sampling from the decoder output distribution
+    and producing the inputs for the next decoding step. In most cases, they should not be
```

Comment:
and produce?

Reply:
I think the structure is correct: "the logic of sampling and producing". But I'm not a native English speaker so not 100% sure.

Reply:
Mmm, I see. It was just a little confusing when I first read it.